.github/workflows/checks.yaml

```
@@ -0,0 +1,38 @@
+name: Checks
```

This seems like an overly generic and non-descriptive name (applies also to workflow file name).

I suggest a more descriptive name, such as `pr-validation.yaml`

```suggestion
name: Validate PR requirements
```

---

Name changed. I prefer to spell out abbreviations, so used "Pull Request".